packages/theme-registry/src/ThemeRegistry.ts

```diff
@@ -0,0 +1,267 @@
+  }
+
+  return registry
+}
+
+/**
+ * Validates the passed registry, if the passed registry is undefined, then it will create an empty registry.
+ * @param registry - the registry to validate
+ * @returns {Registry} the registry
+ * @private
+ */
+function validateRegistry(registry: Registry<RegisteredTheme<BaseTheme>>) {
+  const defaultRegistry = makeRegistry()
+
+  if (typeof registry === 'undefined') {
+    return defaultRegistry
+  }
+
+  let valid = true
+
+  Object.keys(defaultRegistry).forEach((key) => {
+    if (
+      typeof registry[key as keyof Registry<RegisteredTheme<BaseTheme>>] ===
+      'undefined'
+    ) {
+      valid = false
+    }
+  })
+
+  error(valid, '[theme-registry] Invalid global theme registry!')
+
+  return registry
+}
+
+/**
+ * Get the global theme registry.
+ * @return {Registry}  the theme registry
+ * @module getRegistry
+ */
+function getRegistry(): Registry<RegisteredTheme> {
+  return globalThis[GLOBAL_THEME_REGISTRY]
+}
+
+/**
+ * Set the global theme registry.
+ * @param {Registry} registry - the registry to set/replacet the current registry with.
+ * @returns {void}
+ * @module setRegistry
+ */
+function setRegistry(registry: Registry<RegisteredTheme>): void {
+  globalThis[GLOBAL_THEME_REGISTRY] = registry
+}
+
+/**
+ * Clear/reset the global theme registry.
+ * @module clearRegistry
+ * @returns {void}
+ */
+function clearRegistry(): void {
+  setRegistry(makeRegistry())
+}
+
+/**
+ * Get the activated theme object.
+ * @return {RegisteredTheme} the default theme object
+ * @module getCurrentTheme
+ */
+function getCurrentTheme(): RegisteredTheme | undefined {
+  const registry = getRegistry()
+  const { currentThemeKey } = registry
+
+  if (currentThemeKey) {
+    return registry.themes[currentThemeKey]
+  }
+
+  return undefined
+}
+
+/**
+ * Activate a theme by the given themeKey.
+ * @param {String} themeKey - the default theme key
+ * @param {DeepPartial<BaseThemeVariables>} overrides - the overrides for the theme variables
+ * @returns {RegisteredTheme} the registered theme object
+ * @module activateTheme
+ * @private
+ */
+function activateTheme(
+  themeKey: string,
+  overrides: DeepPartial<BaseThemeVariables>
+): RegisteredTheme {
+  const registry = getRegistry()
+  const theme = registry.themes[themeKey]
+
+  if (!theme) {
+    if (themeKey !== DEFAULT_THEME_KEY) {
+      error(
+        theme,
+        `[theme-registry] Could not find theme: '${themeKey}' in the registry.`
+      )
+    }
+  }
+
+  registry.currentThemeKey = themeKey
+  const themeWithOverrides = mergeDeep(
+    theme,
+    overrides
+  ) as RegisteredTheme<BaseTheme>
+  registry.themes[themeKey] = themeWithOverrides
+
+  return themeWithOverrides
+}
+/**
+ * Wraps a theme and provides a method to activate the theme.
+ * @returns the wrapped theme object
+ * @module makeTheme
+ * @private
+ */
+function makeTheme<T extends BaseTheme>(theme: T): RegisteredTheme<T> {
```

Comment:
Do you think we need some checks here that check if the theme object is a valid instui theme format?

Reply:
I moved the `isBaseTheme` from `getTheme.ts` and used that function to validate the theme.

Reply:
Yeah, this one is not easy. I mean you could register themes that don't have all the expected properties, but then our components could behave weirdly because they expect all the properties to be there.

I think I like the check more, and if for some reason we have to delete this check later, then it won't be a breaking change, however if we don't add the check now and have to add it later, then that would be a breaking change.

Reply:
Very too! I prefer having this check too, tbh :)